File: AtlanticClojure/PNML_simplified.clj
( defstruct LocatedElement :location )
( defstruct IdedElement :id :location )
( defstruct URI :value :location )
( defstruct PNMLDocument :xmlns :nets :location )
( defstruct NetElement :type :document :id :location )
( defstruct NetContent :net :location )
( defstruct LabeledElement :location )
( defstruct Label :text :labeledElement :location )
( defstruct Name :location )
( defstruct NetContentElement :net :location :id :location )
( defstruct Arc :source :target :net :location :id :location )
( defstruct Place :net :location :id :location )
( defstruct Transition :net :location :id :location )

File: AtlanticOsloM/PNML_simplified.m
module PNML {

	type LocatedElement {
		Location_Att : Text;
	};

	type IdedElement : LocatedElement {
		Id_Att : Text;
	};

	type URI : LocatedElement {
		Value_Att : Text;
	};

	type PNMLDocument : LocatedElement {
		Xmlns_ContRef : URI;
		Nets_ContRef : NetElement+;
	};

	type NetElement : IdedElement {
		Type_ContRef : URI;
		Document_Ref : PNMLDocument;
		Contents_ContRef : NetContent*;
		Name_ContRef : Name?;
	};

	type NetContent : LocatedElement {
		Net_Ref : NetElement;
		Name_ContRef : Name?;
	};

	type LabeledElement : LocatedElement {
		Labels_ContRef : Label*;
	};

	type Label : LocatedElement {
		Text_Att : Text;
		LabeledElement_Ref : LabeledElement;
	};

	type Name : LabeledElement {
		Net_Ref : NetElement?;
		NetContent_Ref : NetContent?;
	};

	type NetContentElement : NetContent, IdedElement;

	type Arc : NetContent, IdedElement {
		Source_Ref : NetContentElement;
		Target_Ref : NetContentElement;
	};

	type Place : NetContentElement;

	type Transition : NetContentElement;
}